### ISSUE TITLE
Train loss, validate loss not decrease anymore but audio still get better

### ISSUE THREAD

**Issue by EuphoriaCelestial** [NVIDIA/tacotron2#353]
![image](https://user-images.githubusercontent.com/43977734/82413292-45d0a280-9a9f-11ea-9c32-4e37e3aabbf3.png)
When training my model, I have this problem: the loss still go around 0,2-0,3 as you can see in the Tensorboard, but the audio result still get better when continue training.
It get to 0,2 loss very quickly (at around 1000 iters), but the audio is crap at this point, no where near acceptable. And then I leave it for 5-6 days, the loss still is 0,2, but the audio is good. Is it normal? Why is it happen?

**Comment by ksaidin** [comment]
Hi, is this same with my overfitting problem? [>> 344](https://github.com/NVIDIA/tacotron2/issues/344#issue-611692630). I was also getting almost original like outputs and excellent attention image. 

If it is overfitting, that should mean that your model is remembering training dataset. Though I was getting good outputs even from unseen sentences with unseen words, I couldn't be sure how stable my model will be in the future.

**Comment by EuphoriaCelestial** [comment]
> Hi, is this same with my overfitting problem? [>> 344](https://github.com/NVIDIA/tacotron2/issues/344#issue-611692630). I was also getting almost original like outputs and excellent attention image.
> 
> If it is overfitting, that should mean that your model is remembering training dataset. Though I was getting good outputs even from unseen sentences with unseen words, I couldn't be sure how stable my model will be in the future.

I am not sure, it look kinda the same. How much data do you have for training?

**Comment by ksaidin** [comment]
I see, overfitting is not the case then. I was at the ~60th epoch when my validation loss got stuck.

What is your proportion of train and validation files?

**Comment by EuphoriaCelestial** [comment]
I put most of my data to train, with 95% train and 5% validation

**Comment by ksaidin** [comment]
You are training from scratch or using pretrained model from this repo? is your speaker male or female?

Right now I am training male speaker (10hrs) and validation loss is stuck at 0,4

**Comment by EuphoriaCelestial** [comment]
I trained from scratch, with female voice. But I use pre-trained Waveglow.

**Comment by EuphoriaCelestial** [comment]
![ảnh](https://user-images.githubusercontent.com/43977734/82721263-d00a4800-9ce5-11ea-802e-689c53fbdbc1.png)

I caught something strange today, is something wrong?
It happen once after 20-30 iters.

**Comment by CookiePPP** [comment]
@EuphoriaCelestial 
https://github.com/NVIDIA/tacotron2/issues/344#issuecomment-627523412
ksaidin noticed that as well. I don't really know, but your loss still looks normal :shrug:

**Comment by rafaelvalle** [comment]
That spike in grad norm could be caused by noise in your data. You can avoid by clipping the norm of the gradient.
@EuphoriaCelestial Can you share loss curves and attention plots? Please turn "Ignore outliers in chart scaling" on.

**Comment by EuphoriaCelestial** [comment]
this is the tensorboard output of my currently training model:
![ảnh](https://user-images.githubusercontent.com/43977734/85813101-8d2c0c00-b78c-11ea-8999-0b3d613df39b.png)
![ảnh](https://user-images.githubusercontent.com/43977734/85813114-961cdd80-b78c-11ea-84b5-53a6802b5f67.png)
![ảnh](https://user-images.githubusercontent.com/43977734/85813139-a339cc80-b78c-11ea-9fcc-ec0745c733fe.png)
![ảnh](https://user-images.githubusercontent.com/43977734/85813156-acc33480-b78c-11ea-9101-5ddef5eb55aa.png)
![ảnh](https://user-images.githubusercontent.com/43977734/85813170-b51b6f80-b78c-11ea-9aef-3615f9b5dfda.png)
![ảnh](https://user-images.githubusercontent.com/43977734/85813179-b9478d00-b78c-11ea-8406-05c6dc5eb315.png)

**Comment by rafaelvalle** [comment]
It all looks pretty good, including the attention maps. I think the spike comes from some noisy data you have.
If you can't find anything wrong with the data, try clipping the norm of the gradient.
Are you annealing the learning rate?

**Comment by EuphoriaCelestial** [comment]
> Are you annealing the learning rate?

Yes.

**Comment by wangzhenhan93** [comment]
> > Are you annealing the learning rate?
> 
> Yes.

hi, i train on vietnamese speech for 40 epoch and found the sound speed so fast.
[sound.zip](https://github.com/NVIDIA/tacotron2/files/5026728/sound.zip)

**Comment by EuphoriaCelestial** [comment]
@wangzhenhan93 I cant download your file, but this kind of problem mostly cause of mismatch sampling rate in model parameter vs dataset
btw, why do you want to train Vietnamese model? can you please share your dataset?
